### PR TITLE
Named dataset endpoint

### PR DESCRIPTION
**Pull request by gwarmstrong**
This PR adds endpoints for "named datasets" which should allow querying separate alpha/beta/taxonomy/pcoa structures, such as one for 16S Amplicon Sequencing and one for Shotgun metagenomics data, or even some kind of endpoints for qRT-PCR.

Basically duplicates every endpoint except metadata endpoints, and adds a name argument for the dataset.

### CODE REVIEW

File: microsetta_public_api/api/microsetta_public_api.yml [hunk 1]
```diff
@@ -619,6 +619,555 @@ paths:
...
+        - Alpha Diversity
+      summary: Get a Vega schema for alpha diversity cdf matching criteria
+      description: Get a Vega schema for alpha diversity cdf matching criteria
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/metadataQuery'
+      responses:
+        '200':
```
Comment (wasade) [inline microsetta_public_api/api/microsetta_public_api.yml:671]: It looks like this response structure is replicated, should it be a `$ref`?

File: microsetta_public_api/api/microsetta_public_api.yml [hunk 2]
```diff
@@ -619,6 +619,555 @@ paths:
         '404':
           $ref: '#/components/responses/404NotFound'
 
+  '/{dataset}/plotting/diversity/alpha/{alpha_metric}/percentiles-plot':
```
Comment (wasade) [inline microsetta_public_api/api/microsetta_public_api.yml:622]: if `dataset` is now upfront, should the existing endpoints be removed and instead require the interactions to denote the dataset being worked on? "/{dataset}/plotting/..." seems very similar to "/plotting/...", for example.

Comment (gwarmstrong) [reply microsetta_public_api/api/microsetta_public_api.yml:622]: Could get weird if you name a dataset `'plotting'`. Probably solved by the `/dataset/{dataset}` solution you proposed?

Would rather make two passes to add the new endpoints, then remove the old ones.

File: microsetta_public_api/api/microsetta_public_api.yml (hunk 1 above)
Comment (gwarmstrong) [reply microsetta_public_api/api/microsetta_public_api.yml:671]: Eh yeah you're right. There's probably about two dozen other repeated segments throughout here too. I might just be happy to kick the can down the road to a maintenance PR.

Comment (gwarmstrong) [reply microsetta_public_api/api/microsetta_public_api.yml:651]: Fixed in https://github.com/biocore/microsetta-public-api/pull/58/commits/b6060795c459dd3d2ec76223c64a369deb2afce0

Comment (wasade) [reply microsetta_public_api/api/microsetta_public_api.yml:651]: Okay cool, thanks!

File: microsetta_public_api/api/microsetta_public_api.yml (hunk 2 above)
Comment (wasade) [reply microsetta_public_api/api/microsetta_public_api.yml]: :+1:

**Pull request merged**